keps/sig-node/4580-deprecate-kubelet-runonce/README.md

```diff
@@ -0,0 +1,348 @@
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: RemoveRunonce
+  - Components depending on the feature gate: kubelet
+  
+  - Will enabling / disabling the feature require downtime of the control
+    plane? Yes. Flag must be set on kubelet start. To disable, kubelet must be restarted. Hence, there would be brief control component downtime on a given node.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? Yes. See above; disabling would require brief node downtime.
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Using the feature gate is the only way to enable/disable this feature.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+N/A
+
+###### Are there any tests for feature enablement/disablement?
+
+N/A
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+In the alpha `stage`, this feature is disabled by default.
+
+In the beta `stage`, this feature is enabled by default. With this feature enabled, the kubelet will refuse to start if is still using runonce mode.
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+N/A
+
+### Monitoring Requirements
+
+* N/A
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+- \- 2024-04-17: Initial draft KEP
+
+## Drawbacks
+
+## Alternatives
+
```

Comment:
> * Runonce does not work in systemd mode.

Let's document the alternative of fixing this (systemd supports one shot units).

Remember, documenting an alternative is not the same as **selecting** an alternative. We document alternatives to show that we considered them, but they didn't get selected as our preferred approach.

Reply:
This argument holds.

Reply:
This argument holds.

Reply:
Added.